### PR TITLE
Enforce the Minimum Kernel Version 6.3 for UserNamespacesSupport feature

### PR DESCRIPTION
**Pull request by dims**
The KEP calls out the minimum kernel version very explicitly, but we are missing a constant and a check in kubelet to enforce the same.

KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/127-user-namespaces/README.md

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
/hold 

(for reviews)

**Review by AkihiroSuda — APPROVED** [review]

**Comment by dims** [comment]
@AkihiroSuda can you please add `/lgtm` as well?

**Comment by AkihiroSuda** [comment]
/lgtm

(Not a maintainer here though)

**Comment by dims** [comment]
> (Not a maintainer here though)

quite all right, all kubernetes org members can `/lgtm`

File: pkg/kubelet/kubelet_pods.go
Comment (aojea) [inline pkg/kubelet/kubelet_pods.go:145]: This is breaking the kind job that monitors the state of a cluster with alpha features enabled 

https://testgrid.k8s.io/sig-testing-kind#kind-master-alpha-beta

The kubelet fails to start https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kind-alpha-beta-features/1789720580090499072/artifacts/kind-control-plane/kubelet.log

If I understand the feature correctly, the enablement of the feature gate does not necessary means that kernel version is required.
It seems the feature gate is being used as a flag that is incorrect, as if this graduates it will be always enabled, and we can not require `UserNamespacesSupportKernelVersion` for ALL the cluster

@dims @AkihiroSuda can you please relax this validation to check the kernel version supported ONLY when the feature is actually used? 

/cc @liggitt

**Comment by rata** [comment]
I'm just finding out this was added. I think this check is wrong and the warning is wrong too. I'll submit a revert with a complete explanation, but basically RHEL and distros do backport features to old kernels and adding this check completely breaks it. There is still a very clear error returned to the user if the feature can't be supported.

@dims can I ask what you wanted to improve with this compared to what is already happening today?  Because, as I say, the error returned to the user is super clear. I'd argue better than this, that is lost in some node logs only for the admin to see.

**Comment by rata** [comment]
Here is the revert I created, to completely remove the kernel check: https://github.com/kubernetes/kubernetes/pull/130243

**Pull request merged**
